flexmeasures/data/queries/sensors.py

```diff
@@ -65,3 +65,24 @@ def query_sensors_by_proximity(
         closest_sensor_query, account_id
     )
     return closest_sensor_query
+
+
+def query_sensors_by_search_terms(
+    query: Select,
+    search_terms: list[str] | None,
+) -> Select:
+    if search_terms is not None:
+        from flexmeasures.data.models.user import Account
+        from flexmeasures.data.models.time_series import Sensor
+
+        filter_statement = and_(
+            *(
+                or_(
+                    Sensor.name.ilike(f"%{term}%"),
+                    Account.name.ilike(f"%{term}%"),
+                )
+                for term in search_terms
+            )
+        )
+
+    return query.filter(filter_statement)
```

Comment:
If `search_terms` is None, then `filter_statement` is undefined right?

Maybe a small test for this would be good.

Or, to be simpler, move this back into the sensors/index endpoint, now that you are not using this in two endpoints anymore.

Reply:
You mean move this code into sensors API?

Reply:
Yes, that would be easier to read, as well, and I don't believe we'd use many more lines of code, really.

Reply:
Ok, that make sense, we are only using it there too.